p2p/p2p-core/src/client/handshaker/builder.rs

```diff
@@ -60,7 +62,10 @@ impl<N: NetworkZone> HandshakerBuilder<N> {
             address_book: DummyAddressBook,
             core_sync_svc: DummyCoreSyncSvc::static_mainnet_genesis(),
             peer_sync_svc: DummyPeerSyncSvc,
-            protocol_request_svc: DummyProtocolRequestHandler,
+            protocol_request_svc_maker: MapErr::new(
+                Shared::new(DummyProtocolRequestHandler),
+                tower::BoxError::from,
+            ),
```

Comment:
This infallible is being wrapped and returned in an error to satisfy trait bounds, right?